### DISCUSSION TITLE
deprecated virtualScrollBody property in PrimeNG 14

### DISCUSSION THREAD

**Discussion by Sayali77** [primefaces/community#817] [PrimeNG] [answered]
my PrimeNG 13 application, I was utilizing the virtualScrollBody property of the Table component to access the scroll viewport bounds using the following code:

`const scrollViewportBounds = this.table.virtualScrollBody.elementRef.nativeElement.getBoundingClientRect();`

this gives the current scroll location 
However, after upgrading to PrimeNG 14, I've discovered that the virtualScrollBody property is now deprecated, and I'm unable to find a direct replacement. 
Could you please advise on the recommended approach to access the scroll viewport bounds in PrimeNG 14, particularly for virtual scrolling scenarios?
Is there an alternative property or method that I should be using instead of virtualScrollBody?

**Comment by cetincakiroglu** [comment, marked as answer]
Hi,

We've introduced a scroller component in v14 and deprecated the virtualScroller component. Please use scroller instead.
